clustering/process.go

```diff
@@ -230,8 +230,8 @@ func (p *managerProcess) updateStatus(ctx context.Context, ss *StatusSet) error
 
 	now := metav1.Now()
 	ststr := ss.State.String()
-	updateCond := func(typ mocov1beta2.MySQLClusterConditionType, val corev1.ConditionStatus, current []mocov1beta2.MySQLClusterCondition) mocov1beta2.MySQLClusterCondition {
-		updated := mocov1beta2.MySQLClusterCondition{
+	updateCond := func(typ string, val metav1.ConditionStatus, current []metav1.Condition) metav1.Condition {
+		updated := metav1.Condition{
```

Comment:
@zoetrope 
Should I set `ObservedGeneration` here too?

Reply:
It is a difficult question.
process.go and mysqlcluster-controller are asynchronous.
So updated status in process.go are not based on the generation of mysqlcluster.

I think no need to set observedGeneration here.